doc/arkode/guide/source/Usage/LSRKStep/User_callable.rst

```diff
@@ -0,0 +1,303 @@
+      * *nsteps* -- the dominant eigenvalue re-computation update frequency.  A value ``nsteps = 0`` indicates that the dominant eigenvalue will not change throughout the simulation.
+
+   **Return value:**
+      * *ARK_SUCCESS* if successful
+      * *ARKLS_MEM_NULL* if ``arkode_mem`` was ``NULL``.
+      * *ARK_ILL_INPUT* if an argument had an illegal value (e.g. ``nsteps < 0``)
+
+
+.. c:function:: int LSRKStepSetMaxNumStages(void* arkode_mem, int stage_max_limit);
+
+   Specifies the maximum number of stages allowed within each time step.  This bound only applies to
+   RKL and RKC methods, and any inputs are ignored for SSPRK methods.
+
+   **Arguments:**
+      * *arkode_mem* -- pointer to the LSRKStep memory block.
+      * *stage_max_limit* -- maximum allowed number of stages :math:`(>1)`.
+
+   **Return value:**
+      * *ARK_SUCCESS* if successful
+      * *ARKLS_MEM_NULL* if ``arkode_mem`` was ``NULL``.
+      * *ARK_ILL_INPUT* if an argument had an illegal value (e.g. ``stage_max_limit < 2``)
+
+
+.. c:function:: int LSRKStepSetDomEigSafetyFactor(void* arkode_mem, sunrealtype dom_eig_sfty);
+
+   Specifies a safety factor to use for the result of the dominant eigenvalue estimation function.  This value is used to scale the magnitude of the dominant eigenvalue, in the hope of ensuring a sufficient number of stages for the method to be stable.  This input is only used for RKC and RKL methods, and is ignored by SSPRK methods.
+
+   **Arguments:**
+      * *arkode_mem* -- pointer to the LSRKStep memory block.
+      * *dom_eig_sfty* -- safety factor :math:`(\ge 1)`.
+
+   **Return value:**
+      * *ARK_SUCCESS* if successful
+      * *ARKLS_MEM_NULL* if ``arkode_mem`` was ``NULL``.
+      * *ARK_ILL_INPUT* if an argument had an illegal value (e.g. ``dom_eig_sfty < 1``)
+
+
+.. c:function:: int LSRKStepSetSSPStageNum(void* arkode_mem, int num_of_stages);
+
+   Sets the number of stages, ``s`` in ``SSP(s, p)`` methods.  This input is ignored by RKC and RKL methods.
+
+      * ``ARKODE_LSRK_SSP_S_2``  -- ``num_of_stages`` must be greater than or equal to 2
+      * ``ARKODE_LSRK_SSP_S_3``  -- ``num_of_stages`` must be a perfect-square greater than or equal to 9
+      * ``ARKODE_LSRK_SSP_10_4`` -- ``num_of_stages`` cannot be modified from 10, so this function should not be called.
+
+   **Arguments:**
+      * *arkode_mem* -- pointer to the LSRKStep memory block.
+      * *num_of_stages* -- number of stages :math:`(>1)` for ``SSP(s,2)`` and :math:`(n^2 = s \geq 9)` for ``SSP(s,3)``.
+
+   **Return value:**
+      * *ARK_SUCCESS* if successful
+      * *ARKLS_MEM_NULL* if ``arkode_mem`` was ``NULL``.
+      * *ARK_ILL_INPUT* if an argument had an illegal value (e.g. ``num_of_stages < 2``)
+
+
+.. _ARKODE.Usage.LSRKStep.OptionalOutputs:
+
+Optional output functions
+------------------------------
+
+
+.. c:function:: int LSRKStepGetNumRhsEvals(void* arkode_mem, long int* fe_evals, long int* fi_evals);
```

Comment:
I like the second option, since that is extensible to all of the proposed method types.  We'd just need to be careful in the documentation to specify the ordering of the outputs.

Reply:
Agreed, the documentation will need to be clear on the length and ordering depending on the method that's being used.